Source/JavaScriptCore/jsc.cpp

```diff
@@ -1379,6 +1385,39 @@ JSC_DEFINE_HOST_FUNCTION(functionPrettyPrint, (JSGlobalObject* globalObject, Cal
     return printInternal(globalObject, callFrame, stdout, true);
 }
 
+JSC_DEFINE_HOST_FUNCTION(functionAtob, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    auto* string = callFrame->argument(0).toString(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    auto decodedData = base64Decode(string->value(globalObject), { Base64DecodeOptions::ValidatePadding, Base64DecodeOptions::IgnoreSpacesAndNewLines, Base64DecodeOptions::DiscardVerticalTab });
+    if (!decodedData)
+        throwException(globalObject, scope, createError(globalObject, "Invalid argument for atob."_s));
+
+    return JSValue::encode(jsString(vm, String(decodedData->data(), decodedData->size())));
+}
+
+JSC_DEFINE_HOST_FUNCTION(functionBtoa, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    auto* string = callFrame->argument(0).toString(globalObject);
+    RETURN_IF_EXCEPTION(scope, { });
+
+    String stringToEncode = string->value(globalObject);
+
+    if (stringToEncode.isNull())
+        return JSValue::encode(jsString(vm, String()));
```

Comment:
I think you want `jsEmptyString(vm)` instead of `jsString(vm, String())`.  BTW, `String()` is the NULL string as opposed to the empty string.  However, `jsString()` converts it to the empty string.  Using `jsEmptyString(vm)` directly cuts out a lot of unnecessary work in between.